$OpenBSD: patch-beaker_crypto_pycrypto_py,v 1.2 2018/05/12 20:05:25 jasper Exp $

- py-crypto -> py-cryptodome

- Security fix for CVE-2012-3458, py-beaker weak use of crypto can
  leak information to remote attackers.
  Patch from: https://github.com/bbangert/beaker/commit/91becae76101cf87ce8cbfabe3af2622fc328fe5

Index: beaker/crypto/pycrypto.py
--- beaker/crypto/pycrypto.py.orig
+++ beaker/crypto/pycrypto.py
@@ -14,18 +14,20 @@ try:
     aesDecrypt = aesEncrypt
 
 except ImportError:
-    from Crypto.Cipher import AES
+    from Cryptodome.Cipher import AES
+    from Cryptodome.Util import Counter
 
     def aesEncrypt(data, key):
-        cipher = AES.new(key)
+        cipher = AES.new(key, AES.MODE_CTR,
+                         counter=Counter.new(128, initial_value=0))
 
-        data = data + (" " * (16 - (len(data) % 16)))
         return cipher.encrypt(data)
 
     def aesDecrypt(data, key):
-        cipher = AES.new(key)
+        cipher = AES.new(key, AES.MODE_CTR,
+	                 counter=Counter.new(128, initial_value=0))
 
-        return cipher.decrypt(data).rstrip()
+        return cipher.decrypt(data)
 
 def getKeyLength():
     return 32
